File: api/src/main/java/dev/akif/crud/simplest/SimplestRepository.java
package dev.akif.crud.simplest;

import dev.akif.crud.CRUDRepository;

import java.io.Serializable;

/**
 * The simplest variant of {@link CRUDRepository}
 * <p>
 * This is meant to be implemented from a @{@link org.springframework.stereotype.Repository} interface.
 *
 * @param <I> Id type of the data
 * @param <E> Entity type of the data which is a {@link SimplestEntity}
 */
public interface SimplestRepository<I extends Serializable, E extends SimplestEntity<I, E>>
        extends CRUDRepository<I, E> {
}

File: api/src/main/kotlin/dev/akif/crud/simplest/SimplestRepository.kt
package dev.akif.crud.simplest

import dev.akif.crud.CRUDRepository
import java.io.Serializable

/**
 * The simplest variant of [CRUDRepository]
 *
 *
 * This is meant to be implemented from a **@Repository** interface.
 *
 * @param I Id type of the data
 * @param E Entity type of the data which is a [SimplestEntity]
 */
interface SimplestRepository<I : Serializable, E : SimplestEntity<I>> : CRUDRepository<I, E>
